conans/client/build/cmake_toolchain_build_helper.py

```diff
@@ -0,0 +1,253 @@
+import os
+import platform
+
+from conans.client import tools
+from conans.client.build import defs_to_string, join_arguments
+from conans.client.build.cmake import BaseCMakeBuildHelper
+from conans.client.build.cmake_flags import cmake_in_local_cache_var_name
+from conans.client.build.cmake_flags import in_local_cache_definition
+from conans.client.build.cmake_flags import is_multi_configuration, get_generator, \
+    verbose_definition, verbose_definition_name
+from conans.client.toolchain.cmake import CMakeToolchain
+from conans.client.tools.oss import cpu_count, args_to_string
+from conans.errors import ConanException
+from conans.model.version import Version
+from conans.util.config_parser import get_bool_from_text
+from conans.util.files import mkdir
+
+
+def _validate_recipe(conanfile):
```

Comment:
The problem here is that, if you use `cmake` generator you will adjust things in the toolchain and then in `conan_basic_setup`. I'd love to think that they will be the same adjustments and nothing bad will happen (maybe flags will be duplicated), but... I'm not sure if it is safe.

I prefer to keep only one valid scenario, see the issues/bugs related to it and then start to cover other scenarios if needed.... but let's come back to this issue before merging, it would be about removing one function.

Reply:
Agree